drivers/cavium/thunder_mmc.c

```diff
@@ -0,0 +1,425 @@
+/*
```

Comment:
Have Cavium authorised the release of open source drivers for the hardware IP in this contribution, in particular the eMMC, SPI, SATA and TWSI drivers? Are there publicly available data sheets for these IPs?

Reply:
Yes the whole code was release under BSD license.
There's no data sheet publicly available.

Reply:
Released by Cavium? Please provide a link to the source tree

Reply:
It's available here. You have to use git LFS and then install the RPM...
https://github.com/Cavium-Open-Source-Distributions/OCTEON-TX-SDK